android/app/src/main/java/app/organicmaps/settings/SettingsPrefsFragment.java

```diff
@@ -799,10 +801,32 @@ public static ThemeMode getInstance(@NonNull Context context, @NonNull String sr
     }
   }
 
+  private void initSearchPrivacyPrefsCallbacks()
+  {
+    final Preference pref = findPreference(getString(R.string.pref_search_privacy));
+
+    final boolean isSearchPrivacyEnabled = Config.isSearchPrivacyEnabled();
+    ((TwoStatePreference) pref).setChecked(isSearchPrivacyEnabled);
+    pref.setOnPreferenceChangeListener((preference, newValue) -> {
+      boolean newVal = (Boolean) newValue;
+      if (newVal != isSearchPrivacyEnabled)
+      {
+        Config.setSearchPrivacyEnabled(newVal);
+        if (newVal)
+        {
+          SearchRecents.refresh();
+        }
+        else
+        {
+          SearchRecents.clear();
+        }
```

Comment:
No need curly braces here:
```suggestion
        if (newVal)
          SearchRecents.refresh();
        else
          SearchRecents.clear();
```